docarray/array/storage/redis/backend.py

```diff
@@ -0,0 +1,188 @@
+        'long': TypeMap(type='long', converter=NumericField),
+        'bool': TypeMap(type='long', converter=NumericField),
+    }
+
+    def _init_storage(
+        self,
+        _docs: Optional['DocumentArraySourceType'] = None,
+        config: Optional[Union[RedisConfig, Dict]] = None,
+        **kwargs,
+    ):
+        if not config:
+            raise ValueError('Empty config is not allowed for Redis storage')
+        elif isinstance(config, dict):
+            config = dataclass_from_dict(RedisConfig, config)
+
+        if config.distance not in ['L2', 'IP', 'COSINE']:
+            raise ValueError(
+                f'Expecting distance metric one of COSINE, L2 OR IP, got {config.distance} instead'
+            )
+        if config.method not in ['HNSW', 'FLAT']:
+            raise ValueError(
+                f'Expecting search method one of HNSW OR FLAT, got {config.method} instead'
+            )
+
+        if config.redis_config.get('decode_responses'):
+            config.redis_config['decode_responses'] = False
+
+        self._offset2id_key = 'offset2id__' + config.index_name
+        self._config = config
+        self.n_dim = self._config.n_dim
+        self._doc_prefix = "doc__" + config.index_name + ":"
+        self._config.columns = self._normalize_columns(self._config.columns)
+
+        self._client = self._build_client()
+        super()._init_storage()
+
+        if _docs is None:
+            return
+        elif isinstance(_docs, Iterable):
+            self.extend(_docs)
+        elif isinstance(_docs, Document):
+            self.append(_docs)
+
+    def _build_client(self):
+        client = Redis(
+            host=self._config.host,
+            port=self._config.port,
+            **self._config.redis_config,
+        )
+
+        if self._config.flush:
+            client.flushdb()
+
+        if self._config.update_schema:
+            if self._config.index_name in client.execute_command('FT._LIST'):
+                client.ft(index_name=self._config.index_name).dropindex()
+
+        if self._config.flush or self._config.update_schema:
+            schema = self._build_schema_from_redis_config()
+            idef = IndexDefinition(prefix=[self._doc_prefix])
+            client.ft(index_name=self._config.index_name).create_index(
+                schema, definition=idef
+            )
+
+        return client
+
+    def _ensure_unique_config(
+        self,
+        config_root: dict,
+        config_subindex: dict,
+        config_joined: dict,
+        subindex_name: str,
+    ) -> dict:
+        if 'index_name' not in config_subindex:
+            config_joined['index_name'] = (
+                config_joined['index_name'] + '_subindex_' + subindex_name
+            )
+        config_joined['flush'] = False
+        return config_joined
+
+    def _build_schema_from_redis_config(self):
+        index_param = {
+            'TYPE': 'FLOAT32',
```

Comment:
Should this be configurable?

Reply:
TYPE is a mandantory parameter. Current supported type is only FLOAT32. see [here](https://redis.io/docs/stack/search/reference/vectors/)